BackofficeBundle/StrategyManager/AuthorizeStatusChangeManager.php

```diff
@@ -0,0 +1,40 @@
+<?php
+
+namespace OpenOrchestra\BackofficeBundle\StrategyManager;
+
+use OpenOrchestra\Backoffice\AuthorizeStatusChange\AuthorizeStatusChangeInterface;
+use OpenOrchestra\ModelInterface\Event\StatusableEvent;
+
+/**
+ * Class AuthorizeStatusChangeManager
+ */
+class AuthorizeStatusChangeManager
+{
+    protected $strategies = array();
+
+    /**
+     * @param AuthorizeStatusChangeInterface $strategy
+     */
+    public function addStrategy(AuthorizeStatusChangeInterface $strategy)
+    {
+        $this->strategies[$strategy->getName()] = $strategy;
+    }
+
+    /**
+     * @param StatusableEvent $event
+     *
+     * @return bool
+     */
+    public function isGranted(StatusableEvent $statusableEvent)
+    {
+        $granted = true;
+        foreach ($this->strategies as $strategy) {
+            $granted = $granted && $strategy->isGranted($statusableEvent);
```

Comment:
if the return of the strategy is false, you can exit directly with a return;

This way you will limit the work done